### ISSUE TITLE
object ConllChainNer creation failure

### ISSUE THREAD

**Issue by tribbloid** [factorie/factorie#385]
In the following source code in ChainNer.scala

`object ConllChainNer extends ConllChainNer()(ModelProvider.classpath(), StaticLexiconFeatures()) with Serializable`

The ModelProvider.classpath() will erase ClassTag[ChainNer], which leads to the following error:

```
No file named Nothing$.factorie found in classpath for class scala.runtime.Nothing$, and no value found in system property scala.runtime.Nothing$. To fix this either add a file with the right name to the classpath or set the system property to point to a directory containing the file.
java.lang.Error: No file named Nothing$.factorie found in classpath for class scala.runtime.Nothing$, and no value found in system property scala.runtime.Nothing$. To fix this either add a file with the right name to the classpath or set the system property to point to a directory containing the file.
	at cc.factorie.util.ClasspathURL$.apply(ClasspathURL.scala:42)
	at cc.factorie.util.ModelProvider$$anon$1.<init>(ModelProvider.scala:94)
	at cc.factorie.util.ModelProvider$.classpath(ModelProvider.scala:93)
	at cc.factorie.app.nlp.ner.ConllChainNer$.<init>(ChainNer.scala:67)
	at cc.factorie.app.nlp.ner.ConllChainNer$.<clinit>(ChainNer.scala)
```

**CAUTION** this error is NOT caused by model file missing, as evidently indicated by the stacktrace, it is caused by Java/Scala type erasure, in which ClassTag[ChainNer] degrades to ClassTag[Nothing$]

**Comment by strubell** [comment]
@JackSullivan do you know what the problem is here?

**Comment by tribbloid** [comment]
@strubell Just scala 2.11.8 compiler is not smart enough:

I've edit directly on your source code:

`object ConllChainNer extends ConllChainNer()(ModelProvider.classpath(), StaticLexiconFeatures()) with Serializable
`
=>

`object ConllChainNer extends ConllChainNer()(ModelProvider.classpath[ConllChainNer](), StaticLexiconFeatures()) with Serializable`

And it solves the problem.

Not sure if later version of scala became smarter though

**Comment by strubell** [comment]
Ah, great. We'd appreciate it if you could submit the fix as a PR!

**Comment by tribbloid** [comment]
In a few hours.
I wonder why this problem was not discovered before. When was the last big refactoring on NER component again?

**Comment by strubell** [comment]
I'm also surprised this hasn't been found. The last refactor was quite a
while ago, and I have run the NER tagger since that refactor. My guess is
that it's a change in a newer Scala compiler (I probably used ~2.11.4).

**Comment by tribbloid** [comment]
fixed as https://github.com/factorie/factorie/pull/386.
up yours

**Comment by lvilnis** [comment]
not sure what that means ;) but closing as was fixed in #386
